### DISCUSSION TITLE
Check if PKCS11 module is valid or not

### DISCUSSION THREAD

**Discussion by aniljadaun** [Mastercard/pkcs11-tools#58] [Q&A] [answered]
I am new to this. Sorry is advance if my question is too naive.

I have a PKCS11 Module which is not correctly loaded into Acrobat App. Another module is correctly loaded. I need to check if issue is with my module or Acrobat. 

Can you suggest how can i use pkcs11-tools to check if my module is valid ?

**Comment by MrWildanMD** [comment, marked as answer]
can u just do getting slot info from it? or using the native tools provided by the pkcs11 module itself
i.e i'm using utimaco so using csadm could be done csadm GetState and it will be printed the return message
and if using this tools u can do it easily with
i.e utimaco devices (./with_utimaco p11slotinfo -s 0) and u will get also the return message
cheers

**Comment by keldonin** [comment]
As @MrWildanMD suggested, `p11slotinfo` can help you:

```
p11slotinfo -l [PATH_TO_PKCS11_LIBRARY]
```

One caveat however: This will not work on Windows: the toolkit would need to be compiled under MSVC, and that is currently not supported. As the Microsoft compiler uses different conventions for data size than GCC or CLANG under Linux/FreeBSD (namely `unsigned long`), PKCS#11 libraries cannot be reliably used on this platform.
